src/main/resources/avro/common.avdl

```diff
@@ -387,19 +302,59 @@ record Position {
 }
 
 /**
-A `Segment` is a range on a sequence, possibly including the joins at the
-sequence's ends. It does not include base data. (The bases for a sequence are
-available through the `getSequenceBases()` API call.)
+A `Join` is simply a pair of Position objects, which we call the sides. The are
+logically unordered (i.e. swapping makes no difference), but we require a rank
+on the Sequences, and so implicitly on the positions, so to avoid ambiguity we
+require that the position for side1 is less than that for side2.
+*/
+record Join {
```

Comment:
On Wed, Mar 18, 2015 at 9:37 AM, Frank Austin Nothaft <
notifications@github.com> wrote:

> In src/main/resources/avro/common.avdl
> <https://github.com/ga4gh/schemas/pull/250#discussion_r26680976>:
>
> > @@ -387,19 +302,59 @@ record Position {
> >  }
> >
> >  /**
> > -A `Segment` is a range on a sequence, possibly including the joins at the
> > -sequence's ends. It does not include base data. (The bases for a sequence are
> > -available through the `getSequenceBases()` API call.)
> > +A `Join` is simply a pair of Position objects, which we call the sides. The are
> > +logically unordered (i.e. swapping makes no difference), but we require a rank
> > +on the Sequences, and so implicitly on the positions, so to avoid ambiguity we
> > +require that the position for side1 is less than that for side2.
> > +*/
> > +record Join {
>
> I don't know if we *strictly* need a Join type, but I believe that having
> the Join type instead of overloading an array makes the data model
> cleaner.
>

+1 @franknothaft


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ga4gh/schemas/pull/250/files#r26680976>.
>